keps/sig-node/4328-node-label-manager/README.md

```diff
@@ -0,0 +1,442 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+This KEP aims to implement the `RequiredDuringSchedulingRequiredDuringExecution` type for NodeAffinity. This type implies that pods with this affinity rule must meet certain conditions during both the scheduling phase and the execution phase. 
+
+## Motivation
+- Enable strict node affinity for pod scheduling and execution.
+- Provide users with greater control over pod placement.
+- Ensure predictable and consistent pod behavior in various scenarios.
+- Address use cases with specific placement and execution requirements.
+  
+### Goals
+- Introduce the "RequiredDuringSchedulingRequiredDuringExecution" node affinity type.
+- Add a node affinity controller to the controllers to ensure pods being evicted if the selectors are no longer met at runtime.
+
+### Non-Goals
+- This KEP does not affect existing node affinity types.
+- It does not introduce breaking changes for workloads with different requirements.
+- It does not enforce stricter affinity rules for existing workloads.
+- It does not modify core Kubernetes components unrelated to node affinity.
+
+## Proposal
+- Implement the "RequiredDuringSchedulingRequiredDuringExecution" node affinity type.
+- Update the Kubernetes scheduler and relevant components.
+- Define the behavior, syntax, and use cases in documentation.
+- Facilitate testing, adoption, and maintenance of this feature.
+
+### User Stories (Optional)
+
+#### Optimizing Workloads Migrating
+As a Data Scientist, I want to ensure that my GPU-intensive ml workloads are always on nodes with correct type of GPUs during both scheduling and execution. So I labeled one node with "gpu=g1". I added some more powerful nodes, and now I want to move all pods requrieing gpu to this new node. With this feature, I can simply remove the "gpu=g1" from the first node and adds it to the new node, so I can move the target pods and keep the remaining pods intact.
```

Comment:
typo: `requiring`

Reply:
at glance this usecase seems to be nicely covered by descheduler. I'll review the next commit addressing early feedback, perhaps you already covered this point.